### DISCUSSION TITLE
Translation to PT-BR?

### DISCUSSION THREAD

**Discussion by ghost** [openbao/openbao#798] [General]
Hello, I recently started studying the project and would like to collaborate with it. I don’t feel ready for the technical issues yet, so I thought about helping with translating the documentation into PT-BR if that is considered necessary at this moment. I would like to understand the best practice for doing this so as not to cause any trouble. I think the best approach would be to fork the project, update it continuously, and then submit a pull request. Am I correct?

**Comment by cipherboy** [comment]
\o hey @qovzeash -- being a self-admitted stupid American who only knows one language (poorly!), translation is something I've not really considered. I know Gitea used Crowdin, and Mozilla developed Pontoon, but that was about the extent of my knowledge. :-) 

It looks like Docusaurus [supports Crowdin](https://docusaurus.io/), based on their homepage, so it should theoretically be possible to translate our docs.

The core binary will likely never be translatable, in part because it wasn't set up for translations and would be a massive undertaking, but also because many third-party libraries rely on the exact strings returned from error messages/response parameters. As an API-driven project, we'd thus prefer stability over translations as they'd break integrations.

I guess to @JanMa and @DanGhita as well: 

Given at most the docs site and the UI (which isn't directly relied on by third-party integrations but isn't set up for localization right now) would be translatable, is there still value in starting these efforts?

**Reply by ghost** [reply]
Well, the proposal of translation is the first no-technical stuff that I thought to help the project; if you have another way, please let me know. Thank you for the answer; I’m at your disposal.

**Reply by cipherboy** [reply]
@qovzeash Yes, it'd be a very good offer if the project was set up for that type of contribution :-)

We've not inherited a translation effort. Are you comfortable leading such an effort from scratch?

Otherwise, the main documentation site could still use polish in several places, regardless of the language it is in! 

Some random thoughts...

The main page (https://openbao.org/) links to our `CONTRIBUTING.md` file in GitHub in the big "Contributing" section, but the navbar links to the docs page: https://openbao.org/docs/contributing/ -- that latter could also probably be expanded.

The installation guide could probably use some updates: https://openbao.org/docs/install/ -- I think NixOS just merged (https://github.com/NixOS/nixpkgs/pull/354366), Alpine exists, and a few others. 

There's also https://github.com/openbao/openbao/issues/393 if you wanted to take on a bigger reorganization effort :-)

**Reply by ghost** [reply]
@cipherboy for me, it’s okay to lead this effort, but I can contribute to the most important things for now. Can you help identify the main priority? Regarding the ‘Contributing’ section, are you referring to the difference between the ```CONTRIBUTING.md``` file and the section under /docs/contributing?

As for the installation methods, I will check in the next few hours how I can help with this. Thank you for your guidance.

**Reply by cipherboy** [reply]
@qovzeash Yes, the difference between the two contributing files should probably be reconciled and unified.

**Reply by ghost** [reply]
@cipherboy I will work on this in the next weeks.
